### PR TITLE
gives names to the defined gatekeeper permissions

### PR DESCRIPTION
**Pull request by ewollesen**
The aim of this is to prevent accidental errors from typos or the like.

This PR builds on top of #52, so I guess I'll keep it a draft until that is resolved.

### CODE REVIEW

File: clients/gatekeeper.go [hunk 1]
```diff
@@ -51,14 +51,44 @@ type (
...
+//
+// If a future update migrates UserInGroup to return UserPermissions, then
+// PermissionScope could be migrated to a user-defined type, and provide a
+// little more compile-time type checking.
+type PermissionScope = string
+
+const (
+	// PermFollow indicates that the user is permitted to follow another user
+	// via the Care Partner app.
+	PermFollow PermissionScope = "follow"
```
Comment (darinkrauss) [inline clients/gatekeeper.go:75]: [nit] Everywhere else it is "Permission" (non-abbreviated), but using "Perm" just here. Suggest you use non-abbreviated here, too.

**Review by darinkrauss — CHANGES_REQUESTED** [review]
While I think the code is fine (only one change requested), is there other code elsewhere making use of this improvement?

Also, gatekeeper is one of the Node.js services that I'd like to see go away. And, specifically for gatekeeper, I'd like it to be redesigned from the ground-up (the way permissions are handled is unwieldy).

File: clients/gatekeeper.go (hunk 1 above)
Comment (ewollesen) [reply clients/gatekeeper.go:75]: Sure, just thought it was still clear, and saved a few chars. I'll change it.

**Comment by ewollesen** [comment]
> While I think the code is fine (only one change requested), is there other code elsewhere making use of this improvement?

It can't until this exists... Once it exists, then they can use it. It's kinda a downside of the go-common package being separate. I'd like to have made use of it with my recent hydrophone changes though.

> Also, gatekeeper is one of the Node.js services that I'd like to see go away. And, specifically for gatekeeper, I'd like it to be redesigned from the ground-up (the way permissions are handled is unwieldy).

Sure, I could see that. But until then... :D

**Review by darinkrauss — APPROVED** [review]
LGTM!

**Pull request merged**